gravitee-am-repository/gravitee-am-repository-jdbc/src/main/java/io/gravitee/am/repository/jdbc/management/api/JdbcOrganizationUserRepository.java

```diff
@@ -365,9 +370,11 @@ public Flowable<User> findByIdIn(List<String> ids) {
     }
 
     @Override
-    public Maybe<User> findById(ReferenceType referenceType, String referenceId, String userId) {
-        LOGGER.debug("findById({},{},{})", referenceType, referenceId, userId);
-        return userRepository.findById(referenceType.name(), referenceId, userId)
+    public Maybe<User> findById(Reference reference, UserId userId) {
+        Criteria criteria = userIdMatches(userId)
+                .and(referenceMatches(reference));
+        LOGGER.debug("findById({},{})", reference, userId);
```

Comment:
not sure we should userIdMatch here, we should keep the "id" and don't manage the ext/src.